include/seqan3/search/configuration/result_type.hpp

```diff
@@ -0,0 +1,72 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::search_cfg::result_type.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/std/type_traits>
+
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
+#include <seqan3/core/type_traits/template_inspection.hpp>
+#include <seqan3/search/configuration/detail.hpp>
+#include <seqan3/search/search_result.hpp>
+
+namespace seqan3::search_cfg::detail
+{
+
+/*!\brief The configuration element storing the configured search result type.
+ * \ingroup search_configuration
+ *
+ * \tparam search_result_t The type of the search result to capture; must be a type specialisation of
+ *                         seqan3::search_result.
+ *
+ * \details
+ *
+ * Implementation of the search result type configuration element.
+ *
+ * \see seqan3::search_cfg::result_type
+ */
+template <typename search_result_t>
+//!\cond
+    requires seqan3::detail::is_type_specialisation_of_v<search_result_t, search_result>
+//!\endcond
+struct result_type_tag :
+    public pipeable_config_element<result_type_tag<search_result_t>,
+                                   std::type_identity<search_result_t>>
+{
```

Comment:
```suggestion
struct result_type_tag : public pipeable_config_element<result_type_tag<search_result_t>, empty_type>
{
    //!\brief The result type to propagate.
    using search_result_type = search_result_t;
    
```
Since the `value` member will go away sooner or later anyway, wouldn't it be more convenient to just make this empty_type and use a type member variable instead? Then you can use e.g. `search_cfg::result_type::search_result_type` instead of decltyping value and then removing type identity.

If you use `using type = ...` you might even use this as kind of a type trait later on